lib/Service/SmimeService.php

```diff
@@ -89,6 +101,52 @@ public function verifyMessage(string $message): bool {
 		return $valid;
 	}
 
+	/**
+	 * Attempt to extract the signed content from a signed S/MIME message.
+	 * Can be used to extract opaque signed content even if the signature itself can't be verified.
+	 *
+	 * Warning: This method does not attempt to verify the signature.
+	 *
+	 * Requires the openssl extension.
+	 *
+	 * @param string $message Whole message including all headers and parts as stored on IMAP
+	 * @return string Signed content
+	 *
+	 * @throws ServiceException If no signed content can be extracted
+	 */
+	public function extractSignedContent(string $message): string {
+		// Ideally, we should use the more modern openssl cms module as it is a superset of the
+		// smime/pkcs7 module. Unfortunately, it is only supported since php 8.
```

Comment:
We require at least PHP 8 ;)

Reply:
If this is not a blocker I suggest we merge :)